pyqtgraph/graphicsItems/PlotDataItem.py

```diff
@@ -166,7 +166,8 @@ def __init__(self, *args, **kargs):
         self.scatter.sigClicked.connect(self.scatterClicked)
         self.scatter.sigHovered.connect(self.scatterHovered)
 
-        self._viewRangeWasChanged = False
+        self._xViewRangeWasChanged = False
+        self._yViewRangeWasChanged = False
```

Comment:
Replaced generic self._viewRangeWasChanged flag with 
axis specific self._xViewRangeWasChanged and self._yViewRangeWasChanged